resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -272,7 +272,8 @@ private[spark] object Config extends Logging {
 
   val KUBERNETES_ALLOCATION_EXECUTOR_TIMEOUT =
     ConfigBuilder("spark.kubernetes.allocation.executor.timeout")
-      .doc("Time to wait before considering a pending executor timedout.")
+      .doc("Time to wait before a newly created executor POD request, which does not reached " +
+        "the POD pending state yet, considered timedout and will be deleted.")
       .version("3.1.0")
```

Comment:
This came up [in a comment of SPARK-34389](https://issues.apache.org/jira/browse/SPARK-34389?focusedCommentId=17282614&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17282614).